### ISSUE TITLE
ms-appinstaller traps for beginners

### ISSUE THREAD

**Issue by PazerOP** [MicrosoftDocs/msix-docs#176]
This isn't a really well formulated issue, more of just a brain dump of solutions to problems that took me a while to figure out. At the very least, hopefully this issue gets indexed by Google. Perhaps these should be added to https://docs.microsoft.com/en-us/windows/msix/app-installer/troubleshoot-appinstaller-issues ?

I was having issues with `.appinstaller` files that worked fine when downloaded, but failed in certain repeatable ways when trying to reference them via the `ms-appinstaller` protocol (`ms-appinstaller:?source=http://path.to/my/ai.appinstaller`).

* "App installation failed with error message: The parameter is incorrect." (#51) - When using the `ms-appinstaller` protocol, the `source` parameter MUST end in `.appinstaller`. Otherwise you will get this "The parameter is incorrect" error. This has nothing to do with mime types on the server. It also doesn't matter if you redirect to a file ending with `.appinstaller`. Both will fail in the same way. The only way to fix this is to make sure the original url ends with `.appinstaller`.
* "App installation failed with error message: Appinstaller operation failed with error code 0x80072F76. Detail: Unknown error (0x80072f76)" - The `.appinstaller` file MUST have `Content-Length` set. Check to make sure your `Content-Length` header is actually making it through your CDN, if you are using one. I was surprised to find out that Cloudflare strips the `Content-Length` header if your mime type is `application/xml`. In this case, changing my `.appinstaller` mime type to `application/appinstaller` fixed the problem (although it has nothing to do with the specific mime type, `application/something_random_haha` also worked).

Another important thing is to make sure your file host supports multiple entries in the HTTP `Range` header. Amazon S3 does not. Neither does ASP.NET Core 3.1. I ended up writing my own `IActionResult` implementation that supported multiple `Range` entries.

**Comment by glent1** [comment]
I raised #51 

Firstly, thanks for creating this.

I have not yet tried to confirm your assertion about the URL or the multiple range requests, but if you are correct, shame on Microsoft for first introducing such a bug and then making me send them a bunch of diagnostic information and then just closing the issue without working it out.

I will report back when I've tried the fix, which is non-trivial because of all the workarounds I put in place so updating did work.
